lib/trie.js

```diff
@@ -0,0 +1,57 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {}
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function (path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+
+  if (parts.length) {
+    for (var i = 0; i < parts.length; i++) {
+      if (!node || !(parts[i] in node)) {
+        node[parts[i]] = new Trie();
```

Comment:
The first part of the condition will not work, as it distills to `if (!node) node[pars[i]] = new Trie()` - i.e. attempts to assign to an undefined variable.